src/Microsoft.OData.Edm/Csdl/Parsing/SchemaJsonParser.cs

```diff
@@ -0,0 +1,1564 @@
+                        }
+                        else
+                        {
+                            IList<CsdlOperation> operations = TryParseCsdlOperationOverload(propertyName, propertyValue, context);
+                            if (operations != null)
+                            {
+                                csdlOpertions.AddRange(operations);
+                            }
+                            else
+                            {
+                                // The schema object MAY also contain annotations that apply to the schema itself.
+                                // ODL doesn't support the Schema itself annotations. So also reporting them as unexpected member.
+                                ReportUnexpectedElement(propertyValue, context);
+                            }
+                        }
+
+                        break;
+                }
+            });
+
+            CsdlSchema schema = new CsdlSchema(schemaNamespace, alias, version,
+                csdlElements.OfType<CsdlStructuredType>(),
+                csdlElements.OfType<CsdlEnumType>(),
+                csdlOpertions,
+                csdlElements.OfType<CsdlTerm>(),
+                csdlElements.OfType<CsdlEntityContainer>(),
+                outOfLineAnnotations,
+                csdlElements.OfType<CsdlTypeDefinition>(),
+                context.Location());
+
+            if (schema.EntityContainers.Count() > 1)
+            {
+                context.ReportError(EdmErrorCode.JsonSchemaCannotHaveMoreThanOneEntityContainer,
+                    Strings.CsdlJsonParser_SchemaCannotHaveMoreThanOneEntityContainer(context.Path));
+            }
+
+            return schema;
+        }
+
+        #region OutOfLine Annotion
+        // out of line annotations
+
+        /// <summary>
+        ///         * "org.example": {
+        ///  "$Alias": "self",
+        ///  "$Annotations": {
+        ///    "self.Person": {
+        ///      "@Core.Description#Tablet": "Dummy",
+        ///      �
+        ///    }
+        ///  }
+        ///}
+        /// </summary>
+        /// <param name="name"></param>
+        /// <param name="element"></param>
+        /// <param name="context"></param>
+        /// <returns></returns>
+        internal static IList<CsdlAnnotations> ParseCsdlOutOfLineAnnotations(string name, JsonElement element, JsonParserContext context)
+        {
+            Debug.Assert(name == "$Annotations");
+            Debug.Assert(element.ValueKind == JsonValueKind.Object);
+
+            IList<CsdlAnnotations> annotationsCollection = new List<CsdlAnnotations>();
+            element.ParseAsObject(context, (propertyName, propertyValue) =>
+            {
+                // The member name is a path identifying the annotation target
+                string target = propertyName;
+
+                string qualifier = null; // It's form the "target" name, or it's not set again in JSON?

```

Comment:
>  string qualifier = null; // It's form the "target" name, or it's not set again in JSON? [](start = 14, length = 89)

the qualifier is set on the individual annotations, not on the Annotations annotation.

Reply:
So, leave it as "null" is a correct behavior? right?

I am wondering why the syntax of "CsdlAnnotations" need a "qualifier" to construct.

---
In reply to: [443053842](https://github.com/OData/odata.net/pull/1775#discussion_r443053842) [](ancestors = 443053842)